prusti-common/src/vir/to_viper.rs

```diff
@@ -705,3 +711,264 @@ impl<'v> ToViper<'v, Vec<viper::Predicate<'v>>> for Vec<Predicate> {
+        for local_var in self.get_formal_returns() {
+            formal_returns_decl.push(local_var.to_viper_decl(ast));
+        }
+
+        let method = ast.method(
+            &self.name(),
+            &[],
+            &formal_returns_decl,
+            &[],
+            &[],
+            method_body,
+        );
+
+        method
+    }
+}
+
+fn cfg_method_convert_basic_block_path<'v>(
+    cfg_method: &CfgMethod,
+    mut path: Vec<String>,
+    ast: &'v AstFactory,
+    blocks_ast: &mut Vec<viper::Stmt<'v>>,
+    declarations: &mut Vec<viper::Declaration<'v>>,
+) {
+    path.reverse();
+    let mut remaining_blocks: HashMap<_, _> = cfg_method
+        .basic_blocks
+        .iter()
+        .enumerate()
+        .map(|(index, block)| {
+            (
+                index_to_label(&cfg_method.basic_blocks_labels(), index),
+                (index, block),
+            )
+        })
+        .collect();
+    let mut current_label = index_to_label(&cfg_method.basic_blocks_labels(), 0);
+    while let Some((index, block)) = remaining_blocks.remove(&current_label) {
+        blocks_ast.push(block_to_viper(ast, &cfg_method.basic_blocks_labels(), block, index));
+        declarations.push(
+            ast.label(&index_to_label(&cfg_method.basic_blocks_labels(), index), &[])
+                .into(),
+        );
+
+        let mut successors: Vec<_> = block
+            .successor
+            .get_following()
+            .into_iter()
+            .map(|ci| index_to_label(&cfg_method.basic_blocks_labels(), ci.index()))
+            .collect();
+        assert!(!successors.is_empty());
+
+        if successors.len() == 1 {
+            current_label = successors.pop().unwrap();
+        } else if let Some(next_label) = path.pop() {
+            current_label = next_label;
+            assert!(
+                successors.contains(&current_label),
+                "successors: {:?} next_label: {:?}",
+                successors,
+                current_label
+            );
+        } else {
+            break;
+        }
+    }
+
+    for label in config::delete_basic_blocks() {
+        let (index, block) = remaining_blocks.remove(&label).unwrap();
+        let fake_position = Position::default();
+        let stmts: Vec<viper::Stmt> = vec![
+            ast.label(&label, &[]),
+            ast.inhale(
+                ast.false_lit_with_pos(fake_position.to_viper(ast)),
+                fake_position.to_viper(ast),
+            ),
+            successor_to_viper(ast, index, &cfg_method.basic_blocks_labels(), &block.successor),
+        ];
+        blocks_ast.push(ast.seqn(&stmts, &[]));
+        declarations.push(ast.label(&label, &[]).into());
+    }
+
+    for (label, (index, block)) in remaining_blocks {
+        blocks_ast.push(block_to_viper(ast, &cfg_method.basic_blocks_labels(), block, index));
+        declarations.push(ast.label(&label, &[]).into());
+    }
+}
+
+// impl CfgMethod {
```

Comment:
Why is this code commented out?

Reply:
Same reason as above, but this time it is just moved out as a function instead of in impl: https://github.com/viperproject/prusti-dev/blob/294b6820e1d8ebecf0d92a8d2e8c6b013bb393fd/prusti-common/src/vir/to_viper.rs#L775-L844 as this function makes less sense to be added in an impl by itself in the file that contains the CfgMethod definition.